hnn_core/opt_toy_example/optimize_evoked.py

```diff
@@ -0,0 +1,164 @@
+"""
+=================================================
+05. Optimize simulated evoked response parameters
+=================================================
+
+This example demonstrates how to optimize the parameters
+of the model simulation to match an experimental dipole waveform.
+"""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import os.path as op
+
+import matplotlib.pyplot as plt
+
+###############################################################################
+# Let us import hnn_core
+
+import hnn_core
+from hnn_core import (MPIBackend, jones_2009_model, simulate_dipole,
+                      read_dipole)
+
+hnn_core_root = op.join(op.dirname(hnn_core.__file__))
+
+# The number of cores may need modifying depending on your current machine.
+n_procs = 2
+
+###############################################################################
+# First, we will load experimental data into Dipole object.
+#
+# This is a different experiment than the one to which the base parameters were
+# tuned. So, the initial RMSE will be large, giving the optimization procedure
+# a lot to work with.
+
+from urllib.request import urlretrieve
+
+data_url = ('https://raw.githubusercontent.com/jonescompneurolab/hnn/master/'
+            'data/MEG_detection_data/S1_SupraT.txt')
+urlretrieve(data_url, 'S1_SupraT.txt')
+exp_dpl = read_dipole('S1_SupraT.txt')
+
+###############################################################################
+# Now we start the optimization!
+#
+# First, we define a function that will tell the optimization routine how to
+# modify the network drive parameters. The function will take in the Network
+# object with no attached drives, and a dictionary of the paramters we wish to
+# optimize.
+
+
+def set_params(net, params_dict):
```

Comment:
In general, I'm not a big fan of attaching the object type to the name of a variable ... one should use python's object introspection capabilities if necessary + keep object structure as simple as possible so such additional explanation is not necessary
```suggestion
def set_params(net, params):
```